### ISSUE TITLE
Replace admission webhook with a validating admission policy

### ISSUE THREAD

**Issue by mrueg** [prometheus-operator/prometheus-operator#5719]
**What is missing?**
Validating admission policies offer a declarative, in-process alternative to validating admission webhooks using Common Expression Language
https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/

It would be interesting to see if the admissionwebhook could be replaced with a VAP. 

**Why do we need it?**
Simplify usage and deployment as the Validating Admission Webhook becomes a policy that can be adjusted and customized by the user. 

**Environment**

* Prometheus Operator version:

    `Insert image tag or Git SHA here`
    
    

**Anything else we need to know?**:

**Comment by simonpasquier** [comment]
I'm afraid that it will fall short for validating PrometheusRules because of PromQL expression parsing.

**Comment by simonpasquier** [comment]
We also have another issue https://github.com/prometheus-operator/prometheus-operator/issues/5079 about supporting CEL...

**Comment by rexagod** [comment]
Correct me if I'm wrong here, but I envision this to be an additional offering beside the existing web-hook logic, since we already have a thorough validation procedure in place. Once this is in place, users can use VAP for one-off validations if need be, while the heavy-lifting use-cases are still covered by the `pkg/admission` web-hooks.
